database-commons/src/main/java/io/cdap/plugin/db/sink/AbstractDBSink.java

```diff
@@ -114,6 +127,45 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
       validateSchema(collector, driverClass, dbSinkConfig.getTableName(), inputSchema, dbSinkConfig.getDBSchemaName());
     }
   }
+  public void validateOperations(FailureCollector collector, T dbSinkConfig, @Nullable Schema inputSchema) {
+    if (dbSinkConfig.containsMacro(OPERATION_NAME) || dbSinkConfig.containsMacro(RELATION_TABLE_KEY)) {
+      return;
+    }
+    String operation = dbSinkConfig.getOperationName();
+    String relationTableKey = dbSinkConfig.getRelationTableKey();
+    if (operation.equalsIgnoreCase(Operation.INSERT.getName()) && relationTableKey == null) {
+      return;
+    }
+    if (operation.equalsIgnoreCase(Operation.INSERT.getName()) && relationTableKey != null) {
+      collector.addFailure(
+        "Table key must be empty in case of 'Insert'.", null)
+        .withConfigProperty(RELATION_TABLE_KEY).withConfigProperty(OPERATION_NAME);
+      return;
+    }
+    if ((operation.equalsIgnoreCase(Operation.UPDATE.getName()) ||
+      operation.equalsIgnoreCase(Operation.UPSERT.getName())) && relationTableKey == null) {
+      collector.addFailure(
+          "Table key must be set if the operation is 'Update' or 'Upsert'.", null)
+        .withConfigProperty(RELATION_TABLE_KEY).withConfigProperty(OPERATION_NAME);
+      return;
+    }
+    if (inputSchema == null) {
```

Comment:
Test a use case where input schema is null due to any connection property as macro , will it work even then.